src/pages/auth/LoginOkta.tsx

```diff
@@ -0,0 +1,175 @@
+import { gql } from '@apollo/client'
+import { Stack } from '@mui/material'
+import { useFormik } from 'formik'
+import { useEffect, useState } from 'react'
+import { useSearchParams } from 'react-router-dom'
+import styled from 'styled-components'
+import { object, string } from 'yup'
+
+import { Alert, Button, Typography } from '~/components/designSystem'
+import { TextInputField } from '~/components/form'
+import { hasDefinedGQLError } from '~/core/apolloClient'
+import { LOGIN_ROUTE } from '~/core/router'
+import { addValuesToUrlState } from '~/core/utils/urlUtils'
+import { LagoApiError, useFetchOktaAuthorizeUrlMutation } from '~/generated/graphql'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import { useShortcuts } from '~/hooks/ui/useShortcuts'
+import { Card, Page, StyledLogo } from '~/styles/auth'
+
+const getErrorKey = (code: LagoApiError): string => {
+  switch (code) {
+    case LagoApiError.OktaUserinfoError:
+      return 'text_664c98989d08a3f733357f73'
+    case LagoApiError.DomainNotConfigured:
+      return 'text_664c90c9b2b6c2012aa50bd6'
+    default:
+      return 'text_62b31e1f6a5b8b1b745ece48'
+  }
+}
+
+gql`
+  mutation fetchOktaAuthorizeUrl($input: OktaAuthorizeInput!) {
+    oktaAuthorize(input: $input) {
+      url
+    }
+  }
+`
+
+const LoginOkta = () => {
+  const { translate } = useInternationalization()
+  let [searchParams] = useSearchParams()
+  const [errorAlert, setErrorAlert] = useState<LagoApiError>()
+  const [errorField, setErrorField] = useState<LagoApiError>()
+
+  const lagoErrorCode = searchParams.get('lago_error_code')
+
+  const [fetchOktaAuthorizeUrl, { error: fetchOktaAuthorizeUrlError, loading }] =
+    useFetchOktaAuthorizeUrlMutation({
+      context: { silentErrorCodes: [LagoApiError.UnprocessableEntity] },
+      fetchPolicy: 'network-only',
+    })
+
+  useEffect(() => {
+    if (lagoErrorCode) {
+      setErrorAlert(lagoErrorCode as LagoApiError)
+
+      // Remove the error code from the URL, so it disappears on page reload
+      history.replaceState({}, '', window.location.pathname)
```

Comment:
Just observation here. This code is also used in Google SSO code but while writing it I felt it's a bit of a hack and the url could contain other informations that are important for the user experience (got one case iun mind with future SalesForce iFrame usage)

Anyway, if you feel this should be changed or handled differently feel free to do so! Let's discuss about it together if you want 🙌

Reply:
Yep I agree but I thought that clean the url before submitting again would prevent "dirty" url it case if the user is encountering errors multiple times. 
Sometimes it can break with the redirect_uri so I've no strong opinion on this one 🤔